src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RefBasesFromAPI.java

```diff
@@ -0,0 +1,65 @@
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPIMetadata;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.RefAPISource;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceShard;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+
+/**
+ * RefBasesFromAPI queries the Google Genomics API for reference bases that span all of the reads on each shard.
+ *
+ * |---------------- shard 0 ----------------|---------------- shard 0 ----------------|
+ * |-- read a --|   |---- read b ---|           |--------------- read c --------------|
+ *
+ * results in,
+ * |---------- ref bases 1 ---------|           |---------- ref bases 2 --------------|
+ * |-- read a --|   |---- read b ---|           |--------------- read c --------------|
+ *
+ * KV<ref bases 1, [read a, read b]>
+ * KV<ref bases 2, [read c]>
+ */
+public class RefBasesFromAPI {
+    public static PCollection<KV<ReferenceBases, Iterable<GATKRead>>> getBasesForShard(PCollection<KV<ReferenceShard, Iterable<GATKRead>>> reads,
+                                                                                       RefAPISource refAPISource,
+                                                                                       RefAPIMetadata refAPIMetadata) {
+        PCollectionView<RefAPISource> sourceView = reads.getPipeline().apply(Create.of(refAPISource)).apply(View.<RefAPISource>asSingleton());
+        PCollectionView<RefAPIMetadata> dataView = reads.getPipeline().apply(Create.of(refAPIMetadata)).apply(View.<RefAPIMetadata>asSingleton());
+        return reads.apply(ParDo.withSideInputs(sourceView, dataView).of(
+                new DoFn<KV<ReferenceShard, Iterable<GATKRead>>, KV<ReferenceBases, Iterable<GATKRead>>>() {
+                    private static final long serialVersionUID = 1L;
+                    @Override
+                    public void processElement(ProcessContext c) throws Exception {
+                        final ReferenceShard shard = c.element().getKey();
+                        final Iterable<GATKRead> reads = c.element().getValue();
+                        int min = Integer.MAX_VALUE;
+                        int max = 1;
+                        int counter = 0;
+                        for (GATKRead r : reads) {
+                            counter++;
+                            if (r.getStart() < min) {
+                                min = r.getStart();
+                            }
+                            if (r.getEnd() > max) {
+                                max = r.getEnd();
+                            }
+                        }
+                        if (counter == 0) {
+                            return;
+                        }
+                        SimpleInterval interval = new SimpleInterval(shard.getContig(), min, max);
```

Comment:
Agreed. It would be static and take a Locatable iterable. I would call it `getSpanningInterval`, though, because to me "span" already means "minimal cover".  We can also adopt the "fluent" style and just call it `spanning`, so we can write:
```SimpleInterval result = SimpleInterval.spanning(reads);```